File: MadLib.rb
# Mad lib game. This is the link to become familiar with the game https://www.madtakes.com/libs/188.html
# Mad Libs is a phrasal template word game which consists of one player prompting others for a list of words to substitute for blanks in a story before reading aloud.

text = "MadLib"

puts "Welcome to my #{text.upcase} program! Please enter some information below:" # Greeting user. Use upper() function to uppercase madlib
puts "" # Add space between strings

# Array to store all parts of speech
list_of_words = [
  '1 verb ending -ing', 
  '1 adverb',
  '1 adjective',
  '2 verb ending -ing',
  '1 first name',
  '2 first name',
  '3 first name', 
  'noun place', 
  'number between 5 and 80'
]

# Array to store user input
user_inputs = []

# for loop to prompt user inputs
list_of_words.each do |word|
  puts "Please enter #{word}: "
  entering_word = gets
  entering_word = entering_word.chomp
  user_inputs.push(entering_word)
end

puts "#{user_inputs}"


#Print MadLib

puts "I'm #{user_inputs[0].capitalize} a #{user_inputs[1]} #{user_inputs[2]} party for my birthday. I'm #{user_inputs[3]}
my best friends, like #{user_inputs[4].upcase} and #{user_inputs[5].upcase} and #{user_inputs[6].upcase}. The party will be at the
#{user_inputs[7]} Now I am #{user_inputs[8]} old."

File: MadLib.py
# your code goes here


# MadLib Program
text = "madlib"

print(f"Welcome to my {text.upper()} program! Please enter some information below:") # Greeting user. Use upper() function to uppercase madlib
print("") # Add space between strings

# list to store all parts of speech
list_of_words = [
      '1 verb ending -ing', 
      '1 adverb',
      '1 adjective',
      '2 verb ending -ing',
      '1 first name',
      '2 first name',
      '3 first name', 
      'noun place', 
      '2 adjective',
      '1 plural noun',
      'adjective color',
      '2 plural noun', 
      '1 verb',
      '1 noun food',
      '2 noun food',
      '2 verb',
      '3 verb',
      'noun part of the body',
      'noun animal',
      '4 verb',
      '3 plural noun',
      '3 verb ending -ing',
      '4 verb ending -ing',
      '5 verb ending -ing',
      '3 adjective']

print(list_of_words)

# dictionary to store user input

user_inputs = {}

# for loop to prompt user inputs

for word in list_of_words:
  entering_word = input(f'Please enter {word}: ')
  user_inputs[word] = entering_word

print(user_inputs)

# Print MadLib

print(f"""I'm {user_inputs['1 verb ending -ing']}a {user_inputs['1 adverb']} {user_inputs['1 adjective']} party for my birthday. I'm {user_inputs['2 verb ending -ing']}
        my best friends, like {user_inputs['1 first name']} and {user_inputs['2 first name']} and {user_inputs['3 first name']}. The party will be at the
        {user_inputs['noun place']} with {user_inputs['2 adjective']} {user_inputs['1 plural noun']} and {user_inputs['adjective color']} {user_inputs['2 plural noun']} 
        for decorations. First, we will {user_inputs['1 verb']}, like {user_inputs['1 noun food']} and {user_inputs['2 noun food']}, and then we will
        {user_inputs['2 verb']} some party games, like {user_inputs['3 verb']} the {user_inputs['noun part of the body']} on the
        {user_inputs['noun animal']} and {user_inputs['4 verb']} the {user_inputs['3 plural noun']}. Then comes my favorite part: 
        {user_inputs['3 verb ending -ing']} Happy Birthday, {user_inputs['4 verb ending -ing']} presents and {user_inputs['5 verb ending -ing']} some
        {user_inputs['3 adjective']} cake.""")
